From nobody Fri Dec 22 09:58:20 2017
From: James Smart <jsmart2021@gmail.com>
Subject: nvmet_fc: correct broken add_port
Date: Thu, 21 Dec 2017 08:43:35 -0800
References: bsc#1072553
Patch-mainline: Submitted, http://lists.infradead.org/pipermail/linux-nvme/2017-December/014532.html

prior patch to remove the add_port reference also removed the saving
of the port pointer, which mean any io had no port pointer thus was
rejected. Basically nothing works.

Restore the saving of the port pointer.

Signed-off-by: James Smart <james.smart@broadcom.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---

 drivers/nvme/target/fc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/nvme/target/fc.c b/drivers/nvme/target/fc.c
index b652208cb37b..fc22415f84fa 100644
--- a/drivers/nvme/target/fc.c
+++ b/drivers/nvme/target/fc.c
@@ -2522,6 +2522,7 @@ nvmet_fc_add_port(struct nvmet_port *port)
 	list_for_each_entry(tgtport, &nvmet_fc_target_list, tgt_list) {
 		if ((tgtport->fc_target_port.node_name == traddr.nn) &&
 		    (tgtport->fc_target_port.port_name == traddr.pn)) {
+			tgtport->port = port;
 			ret = 0;
 			break;
 		}
-- 
2.13.1
